src/som/compiler/Parser.java

```diff
@@ -1126,8 +1132,16 @@ protected ExpressionNode unaryMessage(final ExpressionNode receiver,
       final boolean eventualSend, final SourceSection sendOperator) throws ParseError {
     SourceCoordinate coord = getCoordinate();
     SSymbol selector = unarySelector();
-    return createMessageSend(selector, new ExpressionNode[] {receiver},
-        eventualSend, getSource(coord), sendOperator);
+
+    if (selector.getString().equals("createPromisePair")) {
```

Comment:
this is now obsolete. See https://github.com/smarr/SOMns/pull/88. The already merged pull requested does that for you automatically and for all other primitives.